sir_cake/seller/statistic.py

```diff
@@ -0,0 +1,135 @@
+
+    items = []
+    categories = []
+    number_items_sold = []
+    total_sales = []
+
+    for basket_item in top_five_basket_items:
+        item = Item.objects.get(id=basket_item['item_to_buy'])
+        items.append(item)
+        categories.append(Item.SHORT_CATEGORY_TO_NAME[item.category])
+        number_sold = basket_item['number_items']
+        number_items_sold.append(number_sold)
+        total_sales.append(number_sold * item.price)
+
+    return zip(items, categories, number_items_sold, total_sales)
+
+
+def _sold_basket_items():
+    return BasketItem.objects.exclude(
+        order__status=Order.Status.PREORDER)
+
+
+def number_customers():
+    return _number_registred_users() + _number_unregistred_users()
+
+
+def last_month_statistic():
+    '''Newly registred users and sold items (€) in last 30days'''
+    time_delta_last_month = timezone.now()-timezone.timedelta(days=30)
+    orders = Order.objects.exclude(status=Order.Status.PREORDER).filter(
+        order_date__gte=time_delta_last_month)
+
+    sales = 0
+    for order in orders:
+        basket_item = BasketItem.objects.filter(order=order)
+        sales += total_order_price(basket_item)
+
+    registred_users = User.objects.filter(
+        date_joined__gte=time_delta_last_month).count()
+
+    return sales, registred_users
+
+
+def _number_registred_users():
+    return User.objects.count()
+
+
+def _number_unregistred_users():
+    return AnonymousUser.objects.count()
+
+
+def sales_status_ratio():
+    order = Order.objects.exclude(status=Order.Status.PREORDER)
+
+    return {'status': [Order.Status.UNSENT, Order.Status.SHIPPED],
+            'quantity': [order.filter(status=Order.Status.PAID).count(),
+                         order.filter(status=Order.Status.SHIPPED).count()]}
+
+
+def sold_per_category():
+    categories_and_sales = {'category': [], 'total_sales': []}
+    category_price_query_set = _sold_basket_items().values(
+        'item_to_buy__category').annotate(value_sold=Sum(
+            F('quantity') * F('item_to_buy__price')))
+
+    for item in category_price_query_set:
+        categories_and_sales['category'].append(
+            Item.SHORT_CATEGORY_TO_NAME[item['item_to_buy__category']])
+        categories_and_sales['total_sales'].append(item['value_sold'])
+    return categories_and_sales
+
+
+def sales_graph():
+    '''Sale trend (€) for last 150 orders'''
+    sales = {'date': [], 'sales': []}
+    orders = Order.objects.exclude(status=Order.Status.PREORDER)[:150]
+    for order in orders:
+        value = total_order_price(BasketItem.objects.filter(order=order))
+        date = order.order_date.strftime("%d.%m.%y")
+
+        try:
+            index = sales['date'].index(date)
+            sales['sales'][index] += value
+        except ValueError:
+            sales['date'].append(date)
+            sales['sales'].append(value)
+    return sales
+
+
+def user_registration_statistic():
```

Comment:
Would it make more sense to count all registrations per day for the last 30 days instead of counting only last 40 users?